lib/dao.js

```diff
@@ -1683,7 +1683,27 @@ DataAccessObject._coerce = function(where, options) {
     }
     var DataType = props[p] && props[p].type;
     if (!DataType) {
-      continue;
+      // Handle nested properties
+      if (p.indexOf('.') !== -1) {
+        var parts = p.split('.');
```

Comment:
Please use `const` instead of `var` in the new code, see http://loopback.io/doc/en/contrib/style-guide.html#indentation-of-multi-line-expressions-in-if